future-apps/payroll/test/contracts/Payroll_rates.test.js

```diff
@@ -38,12 +37,338 @@ contract('Payroll rates handling,', ([owner, employee]) => {
+        assert.equal(currentETH.plus(txCost).toString(), previousETH.toString(), 'expected current ETH amount does not match')
+
+        // no DAI income expected
+        const currentDAI = await DAI.balanceOf(employee)
+        assert.equal(currentDAI.toString(), previousDAI.toString(), 'expected current DAI amount does not match')
+
+        // expected an income of 240 ANT since we accrued 2 minutes of salary at 1 USD per second, and the ANT rate is 0.5 USD
+        const currentANT = await ANT.balanceOf(employee)
+        assert.equal(currentANT.minus(previousANT).toString(), bigExp(240, 18).toString(), 'expected current ANT amount does not match')
+      })
+    })
+
+    context('when the employee requests multiple tokens', () => {
+      beforeEach('set token allocations', async () => {
+        await payroll.determineAllocation([ETH, DAI.address, ANT.address], [50, 25, 25], { from: employee })
+      })
+
+      it('receives the expected amount of tokens', async () => {
+        const previousETH = await web3.eth.getBalance(employee)
+        const previousDAI = await DAI.balanceOf(employee)
+        const previousANT = await ANT.balanceOf(employee)
+
+        const { tx, receipt } = await payroll.payday(PAYMENT_TYPES.PAYROLL, 0, { from: employee })
+        const { gasPrice } = await web3.eth.getTransaction(tx)
+        const txCost = gasPrice.mul(receipt.gasUsed)
+
+        // expected an income of 3 ETH having 50% allocated, since we accrued 2 minutes of salary at 1 USD per second, and the ETH rate is 20 USD
+        const currentETH = await web3.eth.getBalance(employee)
+        assert.equal(currentETH.minus(previousETH).plus(txCost).toString(), bigExp(3, 18).toString(), 'expected current ETH amount does not match')
+
+        // expected an income of 30 DAI having 25% allocated, since we accrued 2 minutes of salary at 1 USD per second, and the DAI rate is 1 USD
+        const currentDAI = await DAI.balanceOf(employee)
+        assert.equal(currentDAI.minus(previousDAI).toString(), bigExp(30, 18).toString(), 'expected current DAI amount does not match')
+
+        // expected an income of 60 ANT having 25% allocated, since we accrued 2 minutes of salary at 1 USD per second, and the ANT rate is 0.5 USD
+        const currentANT = await ANT.balanceOf(employee)
+        assert.equal(currentANT.minus(previousANT).toString(), bigExp(60, 18).toString(), 'expected current ANT amount does not match')
+      })
+    })
+  })
+
+  context('when the denomination token is ETH', async () => {
+    beforeEach('initialize payroll app', async () => {
+      ({ payroll, priceFeed } = await createPayrollAndPriceFeed(dao, payrollBase, owner, NOW))
+      await payroll.initialize(finance.address, ETH, priceFeed.address, RATE_EXPIRATION_TIME, { from: owner })
+    })
+
+    beforeEach('set rates and allow tokens', async () => {
+      await setTokenRate(priceFeed, DAI, ETH, DAI_RATE.div(ETH_RATE)) // 0.050 ETH
```

Comment:
What do you think about keeping the base token in the price feed as `ETH`? I think this would be a better test since we've already covered the "inverse" case above with USD being the quote token in the price feed.

Reply:
sure, I think it make sense :)